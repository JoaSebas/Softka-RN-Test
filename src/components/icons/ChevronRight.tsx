import React from 'react';
import Svg, {Path} from 'react-native-svg';
import {colors} from '../../styles/colors';

const ChevronRight = ({size = 24, color = colors.icons}) => {
  return (
    <Svg testID="chevron-right-icon"  width={size} height={size} viewBox="0 0 24 24" fill="none">
      <Path
        d="M9 6L15 12L9 18"
        stroke={color}
        strokeWidth={2}
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </Svg>
  );
};

export default ChevronRight;
