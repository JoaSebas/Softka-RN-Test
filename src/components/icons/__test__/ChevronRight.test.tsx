import React from 'react';
import { render } from '@testing-library/react-native';
import ChevronRight from '../ChevronRight';

describe('ChevronRight Icon', () => {
  it('should render correctly', () => {
    const { getByTestId } = render(<ChevronRight />);
    expect(getByTestId('chevron-right-icon')).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = render(<ChevronRight size={32} color="red" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
