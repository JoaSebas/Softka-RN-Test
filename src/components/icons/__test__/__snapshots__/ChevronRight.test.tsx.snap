// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronRight Icon should match snapshot 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={32}
  bbWidth={32}
  fill="none"
  focusable={false}
  height={32}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 32,
        "width": 32,
      },
    ]
  }
  testID="chevron-right-icon"
  vbHeight={24}
  vbWidth={24}
  width={32}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      d="M9 6L15 12L9 18"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
          "strokeWidth",
          "strokeLinecap",
          "strokeLinejoin",
        ]
      }
      stroke={
        {
          "payload": 4294901760,
          "type": 0,
        }
      }
      strokeLinecap={1}
      strokeLinejoin={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
